python/salobj/test_utils.py

```diff
@@ -131,26 +137,38 @@ def do_setArrays(self, id_data):
 
     def do_setScalars(self, id_data):
         """Execute the setScalars command."""
+        if self.state != base_csc.State.ENABLED:
+            raise utils.ExpectedError(f"setArrays not allowed in {self.state} state")
         self.copy_scalars(id_data.data, self.evt_scalars)
         self.copy_scalars(id_data.data, self.tel_scalars)
         self.controller.evt_scalars.put(self.evt_scalars, 1)
         if not self._enable_scalars_telemetry:
             self._enable_scalars_telemetry = True
             asyncio.ensure_future(self.run_scalars_telemetry())
 
+    def do_fault(self, id_data):
+        """Execute the fault command.
+
+        Change the summary state to State.FAULT
+        """
+        self.fault()
+
     def do_wait(self, id_data):
-        """Execute the wait command."""
-        async def end_wait(id_data):
-            await asyncio.sleep(id_data.data.duration)
-            ack = self.controller.cmd_wait.AckType()
-            ack.ack = id_data.data.ack
-            self.controller.cmd_wait.ack(id_data, ack.ack, 0, "")
-
-        asyncio.ensure_future(end_wait(id_data))
+        """Execute the wait command.
+
+        Wait for the specified time and then acknowledge the command
+        using the specified ack code.
+        """
+        asyncio.ensure_future(self._impl_wait(id_data))
         ack = self.controller.cmd_wait.AckType()
-        ack.ack = self.salinfo.lib.SAL__CMD_INPROGRESS
+        ack.ack = self.controller.salinfo.lib.SAL__CMD_INPROGRESS
         return ack
```

Comment:
It took me a little while to understand the fact that `do_wait` returns an `ack` when no other `do` method return anything. Wouldn't it be better to add a `return None` to all the `do` methods so it is more explicit that the method should actually return something?

Reply:
I'm sorry you found it confusing, but I feel that explicit "return None" adds clutter that many Python programmers would find disturbing. I think I should clarify the documentation, instead.

As an aside, I am becoming convinced that allowing `do_<name>` to be an asynchronous function will really simplify some code. If I can manage that, then I doubt there will be any need to support returning an Ack from a `do_<name>` method.